olp-cpp-sdk-dataservice-read/src/repositories/CatalogRepository.cpp

```diff
@@ -184,6 +184,96 @@ CatalogVersionResponse CatalogRepository::GetLatestVersion(
   return version_response;
 }
 
+client::CancellationToken CatalogRepository::GetLatestVersion(
+    const CatalogVersionRequest& request, CatalogVersionCallback callback) {
+  repository::CatalogCacheRepository repository(
+      catalog_, settings_.cache, settings_.default_cache_expiration);
+
+  const auto fetch_option = request.GetFetchOption();
+  // in case get version online was never called and version was not found in
+  // the cache.
+  CatalogVersionResponse version_response = {
+      {client::ErrorCode::NotFound, "Failed to find a catalog version."}};
+
+  auto func = [=, &request,
+               &version_response]() mutable -> CatalogVersionResponse {
```

Comment:
You capture `version_response` by reference and also return a value of the same type - `CatalogVersionResponse`. Do you really need these two different approaches to return a result from the lambda?